compiler/resolution/functionResolution.cpp

```diff
@@ -7601,6 +7639,17 @@ static void instantiate_default_constructor(FnSymbol* fn) {
         }
       }
     }
+
+    if (AggregateType* ct = toAggregateType(fn->_this->type)) {
+      if (ct->instantiationStyle == DEFINES_INITIALIZER &&
```

Comment:
Definitely.  It happens today so that constructors ensure the fields have been given an initial value before the user constructor assigns into them.  With Phase 1, I hope that won't be necessary (and if it is, I have problems).